connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -342,7 +343,11 @@ private PluginScanResult scanPluginPath(
     }
 
     private static <T>  String versionFor(T pluginImpl) {
-        return pluginImpl instanceof Versioned ? ((Versioned) pluginImpl).version() : "undefined";
+        return pluginImpl instanceof Versioned ? ((Versioned) pluginImpl).version() : UNDEFINED_VERSION;
+    }
+
+    private static <T> String versionFor(Class<? extends T> pluginKlass) throws IllegalAccessException, InstantiationException {
+        return pluginKlass.isAssignableFrom(Versioned.class) ? versionFor(pluginKlass.newInstance()) : UNDEFINED_VERSION;
```

Comment:
It's a bit subtle that currently the only implementation of `Versioned` are `Connector` subclasses. Perhaps it'd be useful to add a comment here that says as much.

Or, because we can't really control other classes (accidentally) implementing `Versioned` in the future, is it more important here to keep checking for assignability from `Connector` rather than `Versioned`? I actually like that, since it matches previous behavior and it insulates us from accidental use